### PR TITLE
Upgrade packages and improve dependency management

### PR DESCRIPTION
**Pull request by jtherrmann**
This PR:

* Removes sub-dependency version pins
* Upgrades all packages to latest versions

TODO:

- [x] Add a `dependabot.yml` file

### CODE REVIEW

**Comment by forrestfwilliams** [comment]
@jtherrmann this all looks good to me, but I don't think I have the full context needed to give an approval. I'll wait for someone else to give the thumbs up.

**Comment by cirrusasf** [comment]
The branch passed the unit test.

File: verify/src/metadata_schema.json [hunk 1]
```diff
@@ -166,7 +166,7 @@
       "type": "array",
       "minItems": 2,
       "maxItems": 2,
-      "items": [
+      "prefixItems": [
```
Comment (asjohnston-asf) [inline verify/src/metadata_schema.json:169]: Looks great, just linking the relevant jsonschema changelog entry here for posterity: https://json-schema.org/draft/2020-12/release-notes.html#changes-to-items-and-additionalitems

Comment (asjohnston-asf) [inline verify/src/metadata_schema.json:169]: Looks great, just linking the relevant jsonschema changelog entry here for posterity: https://json-schema.org/draft/2020-12/release-notes.html#changes-to-items-and-additionalitems

**Review by asjohnston-asf — APPROVED** [review]

**Pull request merged**